gauged/writer.py

```diff
@@ -175,6 +175,36 @@ def clear_from(self, timestamp):
             raise ValueError('Timestamp must be on a block boundary')
         self.driver.clear_from(offset, timestamp)
 
+    def clear_key_before(self, key, namespace=None, timestamp=None):
+        '''Clear all data before `timestamp` for a given key. Note that the timestamp
+        is rounded down to the nearest block boundary'''
+        block_size = self.config.block_size
+        if namespace is None:
+            namespace = self.config.namespace
+        if timestamp is not None:
+            offset, remainder = divmod(timestamp, block_size)
+            if remainder:
+                raise ValueError('timestamp must be on a block boundary')
+            if offset > 0:
+                offset -= 1
```

Comment:
At the moment if you call `writer.clear_key_before(key, timestamp=0)` it will delete the data between 0 and the first block offset (e.g. `[0, 86399]`) if you're using a daily block size. This is probably not what the user expects.